### PR TITLE
Implement suggested fix for 3448

### LINKED ISSUE #3448: Markdown variables consume too many characters if parentheses are nested with whitespace before final )
Not sure what is causing this issue, but it is a bug of some sort.

Some combination of links and parentheses inside tables merges cells together, sometimes splitting words.

the following code will work as a demonstration:
```
|  title 1  | title 2 | title 3 | title 4|
|-----------|---------|---------|--------|
|[foo](bar) |  Ipsum  |    )    |   )    |
```
results in this:
image

### PR DESCRIPTION
**Pull request by dbolack-ab**
## Description

Per issue, remove trim sequence and remove unneeded if wrapper.

```
|  title 1  | title 2 | title 3 | title 4|
|-----------|---------|---------|--------|
|[foo](bar) |  Ipsum  |    )    |   )    |
```

image

## Related Issues or Discussions

- Closes #3448

### Reviewer Checklist

_Replace the list below with specific features you want reviewers to look at._

*Reviewers, refer to this list when testing features, or suggest new items *
- [ ] Verify new features are functional
  - [ ] Test as above

### CODE REVIEW

File: shared/naturalcrit/markdown.js [hunk 1]
```diff
@@ -680,7 +681,7 @@ function MarkedVariables() {
 					}
 					if(match[8]) { // Inline Definition
 						const label = match[10] ? match[10].trim().replace(/\s+/g, ' ') : null; // Trim edge spaces and shorten blocks of whitespace to 1 space
-						let content = match[11] ? match[11].trim().replace(/\s+/g, ' ') : null; // Trim edge spaces and shorten blocks of whitespace to 1 space
+						let content = match[11] ? match[11] : null; // Trim edge spaces and shorten blocks of whitespace to 1 space
```
Comment (calculuschild) [inline shared/naturalcrit/markdown.js:684]: We can remove the comment since we aren't trimming here anymore.

Also I realize now we can just use:

```suggestion
						let content = match[11] || null;
```

**Review by calculuschild — CHANGES_REQUESTED** [review]
Since this fixes a new edge case in our Markdown, we should also add this as a test in our variables tests, to make sure we don't break it in the future.

Otherwise looks pretty straightforward.

File: tests/markdown/variables.test.js [hunk 2]
```diff
@@ -402,4 +402,10 @@ describe('Variable names that are subsets of other names', ()=>{
 		const rendered = Markdown.render(source).trimReturns();
 		expect(rendered).toBe('<p>14</p>');
 	});
+});
+
+describe('Don\'t Eat all the parentheticals!', ()=>{
+	const source='\n|  title 1  | title 2 | title 3 | title 4|\n|-----------|---------|---------|--------|\n|[foo](bar) |  Ipsum  |    )    |   )    |\n';
+	const rendered = Markdown.render(source).trimReturns();
+	expect(rendered).toBe('<table><thead><tr><th>title 1</th><th>title 2</th><th>title 3</th><th>title 4</th></tr></thead><tbody><tr><td><a href=\"bar\">foo</a></td><td>Ipsum</td><td>)</td><td>)</td></tr></tbody></table>');
```
Comment (calculuschild) [inline tests/markdown/variables.test.js:410]: Tests need to be inside an `it()` call. `describe()` is a grouping that contains the individual `it()` tests, but the `describe()` isn't a test itself.

**Comment by calculuschild** [comment]
Alright, I can see the test passing now. Ready to merge!
